src/main/java/com/rocketden/main/service/GameManagementService.java

```diff
@@ -234,4 +236,16 @@ public void updateCode(String roomId, String userId, PlayerCode playerCode) {
     protected boolean isGameOver(Game game) {
         return game.getAllSolved() || (game.getGameTimer() != null && game.getGameTimer().isTimeUp());
     }
+
+    // Update people's socket active status
+    public void conditionallyUpdateSocketInfo(Room room, User user) {
+        Game game = currentGameMap.get(room.getRoomId());
+
+        if (game != null) {
+            log.info("Updating socket info for game {}", room.getRoomId());
+            game.setRoom(room);
+            game.getPlayers().get(user.getUserId()).setUser(user);
```

Comment:
This line actually causes a bug - see my screencast [here](https://www.loom.com/share/72644632b56843ae8b8c5566c381268c).

In short, when you create and end a game, then "Play Again", the game still exists with that room but the players aren't updated. Thus, when the game tried to get the player with that particular user ID, it returns NULL, and then the code tries to run `setUser` on NULL, causing a Null Pointer Exception. This makes the socket update not send (though oddly still shows the user as connected on the host's side) - do you know why this may be?

You _could_ add a test for this specific case, but I'll leave that up to you - I'll approve this after even if you don't add that.

In the future, when should we delete the "game" in the backend? When the "room" is deleted from the database, or right after the game ends each time (transforming it into some sort of history object in the database)?

Reply:
Good catch - fixed! Also added a test case for it

As for when games are deleted - I'm not sure, I think when the room is deleted makes sense, or maybe after play again is clicked. The current design needs the game object to exist in order to show the results page, but if in the future we had like history objects or something then yeah maybe we could also delete them immediately after the game ends (and history is saved).